pkg/controllers/subnetbinding/subnetbinding_controller.go

```diff
@@ -338,22 +338,38 @@ func (r *Reconciler) validateVpcSubnetsBySubnetCR(ctx context.Context, namespace
 
 	// Check if the Subnet CR is nested.
 	if !isTarget {
-		bms := r.SubnetBindingService.GetSubnetConnectionBindingMapsByParentSubnet(subnetPaths[0])
+		bms, err := r.getSubnetConnectionBindingMapsByParentSubnet(ctx, namespace, name)
```

Comment:
If the purpose is to check the CR instead of the NSX resource, should not move this validation to the beginning? I mean this validation has no dependency on the existence of `subnetPaths`, it can be performed earlier.